### PR TITLE
Add benchmark for collation

### LINKED ISSUE #5500: Add benchmark test for collation
## Enhancement

Add benchmark test for collation

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: close #5500

Problem Summary:

### What is changed and how it works?

Benchmark result:
```
------------------------------------------------------------------------------------------
Benchmark                                                Time             CPU   Iterations
------------------------------------------------------------------------------------------
CollationBench/UTF8MB4_BIN/iterations:10           7813733 ns      7813741 ns           10
CollationBench/UTF8MB4_GENERAL_CI/iterations:10   46884226 ns     46872591 ns           10
CollationBench/UTF8MB4_UNICODE_CI/iterations:10  198610375 ns    198593682 ns           10
CollationBench/UTF8_BIN/iterations:10              7695599 ns      7695574 ns           10
CollationBench/UTF8_GENERAL_CI/iterations:10      46136461 ns     46136230 ns           10
CollationBench/UTF8_UNICODE_CI/iterations:10     197835393 ns    197784332 ns           10
CollationBench/ASCII_BIN/iterations:10             9970229 ns      9970250 ns           10
CollationBench/BINARY/iterations:10                9882461 ns      9882442 ns           10
CollationBench/LATIN1_BIN/iterations:10            9768813 ns      9768818 ns           10
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Functions/tests/bench_collation.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+namespace DB
+{
+namespace tests
+{
+
+class CollationBench : public benchmark::Fixture
+{
+public:
+    void SetUp(const benchmark::State &) override {
+        data.push_back(toVec<String>("col1", std::vector<ColStringType>(1000000, "aaaaaaaaaaaaa")));
```
Comment (ywqzzy) [inline dbms/src/Functions/tests/bench_collation.cpp:31]: maybe add more data?

Comment (xzhangxian1008) [reply dbms/src/Functions/tests/bench_collation.cpp:31]: > 

It's enough to show performance difference between collations.

File: dbms/src/Functions/tests/bench_collation.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Functions/FunctionsComparison.h>
+#include <TestUtils/FunctionTestUtils.h>
+#include <benchmark/benchmark.h>
+
+#include <random>
```
Comment (ywqzzy) [inline dbms/src/Functions/tests/bench_collation.cpp:19]: remove this include

Comment (xzhangxian1008) [reply dbms/src/Functions/tests/bench_collation.cpp]: > remove this include

okk

**Review by ywqzzy — APPROVED** [review]
LGTM

File: dbms/src/Functions/tests/bench_collation.cpp (hunk 1 above)
Comment (JaySon-Huang) [reply dbms/src/Functions/tests/bench_collation.cpp:29]: Can you show the current performance result of different collations?

Comment (xzhangxian1008) [reply dbms/src/Functions/tests/bench_collation.cpp:29]: > Can you show the current performance result of different collations in this PR?

okk

**Review by windtalker — APPROVED** [review]
LGTM

**Comment by windtalker** [comment]
Better to add a test for `like`

**Pull request merged**